### PR TITLE
ISPN-8896 Fix HTTP/2 support

### PR DESCRIPTION
**Pull request by slaskawi**
https://issues.jboss.org/browse/ISPN-8896

### CODE REVIEW

File: server/rest/src/test/java/org/infinispan/rest/Http2Test.java [hunk 1]
```diff
@@ -41,6 +41,7 @@
  *
  * @author Sebastian Łaskawiec
  */
+@Test(groups = "functional", testName = "rest.Http2Test")
```
Comment (slaskawi) [inline server/rest/src/test/java/org/infinispan/rest/Http2Test.java]: This is silly... why TestNG doesn't pick up classes ending with `test`, just like JUnit? :imp:

**Comment by slaskawi** [comment]
None of the failures is related.

**Comment by gustavonalle** [comment]
@slaskawi Can we come up with a test for the HTTP/2 support? It was broken as we did not realize it :disappointed: 

Maybe pick a non-netty client? There is a couple aroung, e.g.  http://unrestful.io/2015/10/10/http2-java-client-examples.html

**Comment by slaskawi** [comment]
> @slaskawi Can we come up with a test for the HTTP/2 support? It was broken and we did not realize it disappointed

It was actually my fault. I didn't annotate the test with `@Test`... It's so silly and confusing that this works out of the box for JUnit and doesn't for TestNG...

**Comment by gustavonalle** [comment]
@slaskawi right, I can see that  ```Http2Test``` is failing on master when running from the IDE  :+1:

**Comment by gustavonalle** [comment]
merged, thanks!

**Pull request merged**